modin/engines/base/io/text/csv_dispatcher.py

```diff
@@ -159,16 +168,54 @@ def _read(cls, filepath_or_buffer, **kwargs):
                 **partition_kwargs,
             }
 
+            if pre_reading:
+                pre_reading = {"rows_number": pre_reading}
+                pre_reading["start"] = f.tell()
+                outside_quotes, read_rows, _ = cls._read_rows(
+                    f,
+                    nrows=pre_reading["rows_number"],
+                    quotechar=quotechar,
+                    is_quoting=is_quoting,
+                )
+                read_rows_counter += read_rows
+
+                # add outside_quotes
+                if is_quoting and not outside_quotes:
+                    warnings.warn("File has mismatched quotes")
+
+                pre_reading["end"] = f.tell()
+
+            if isinstance(skiprows, int):
+                read_rows_counter += rows_skipper(skiprows)
+
             splits = cls.partitioned_file(
                 f,
                 num_partitions=num_partitions,
                 nrows=nrows,
-                skiprows=skiprows,
+                skiprows=skiprows if should_handle_skiprows else None,
+                pre_reading=pre_reading,
+                read_rows_counter=read_rows_counter - 1
+                if kwargs.get("encoding", None) is not None
+                else read_rows_counter,
                 quotechar=quotechar,
                 is_quoting=is_quoting,
+                partition_skiprows=1
+                if kwargs.get("encoding", None) is not None
+                else None,
             )
-            for start, end in splits:
-                args.update({"start": start, "end": end})
+            for split_data in splits:
+                rows_considered = split_data[3]
+                # rows_considered == 0 means that partition will contain empty DataFrame
+                # and in this case index_col parameter correction made previously,
+                # should be canceled
+                args["index_col"] = None if rows_considered == 0 else args["index_col"]
```

Comment:
can we just do?
```python
if rows_considered == 0:
    continue
```

Reply:
this approach with modifying `args` dict fails when one of the middle partitions is empty: if we set `args["index_col"] = None` for a single partition, then all partitions after that will also get `None` in this argument

Reply:
Yes, right.